torchtitan/distributed/parallel_dims.py

```diff
@@ -56,13 +62,24 @@ def build_mesh(self, device_type):
             [self.pp, self.dp_replicate, self.dp_shard, self.cp, self.tp],
             ["pp", "dp_replicate", "dp_shard", "cp", "tp"],
         ):
-            if d > 1:
+            if d > 1 or (name == "dp_replicate" and self.ft_manager is not None):
                 dims.append(d)
                 names.append(name)
 
         logger.info(f"Building {len(dims)}-D device mesh with {names}, {dims}")
         names = tuple(names)
-        mesh = init_device_mesh(device_type, dims, mesh_dim_names=names)
+        if self.ft_manager is None:
```

Comment:
wondering if there is a way to move the ft_manager related logic to do post processing on `ParallelDims` after creating it? The reason I ask is because torchft might be confusing as an argument to `ParallelDims` for users reading the code since its only tangentially related to parallelism.

Reply:
That's actually a very good question. However it is not easy to do that as we need the information of TorchFT.manager to initialize DeviceMesh. 

Another approach that I discussed with @tianyu-l is that we can probably create `ParallelDimsFT` so that users who don't care about FT can easily understand the original `ParallelDims` without any FT knowledge.

I'll probably do this in the next PR.

Reply:
I decided to add `FTParallelDim` in this PR.